exercises/chapter3/src/Data/AddressBook.purs

```diff
@@ -1,40 +1,70 @@
+-- ANCHOR: imports
 module Data.AddressBook where
 
 import Prelude
 
 import Control.Plus (empty)
 import Data.List (List(..), filter, head)
 import Data.Maybe (Maybe)
+-- ANCHOR_END: imports
 
+-- ANCHOR: Address
 type Address =
   { street :: String
   , city   :: String
   , state  :: String
   }
+-- ANCHOR_END: Address
 
+-- ANCHOR: Entry
 type Entry =
   { firstName :: String
   , lastName  :: String
   , address   :: Address
   }
+-- ANCHOR_END: Entry
 
+-- ANCHOR: AddressBook
 type AddressBook = List Entry
+-- ANCHOR_END: AddressBook
 
+-- ANCHOR: showAddress
 showAddress :: Address -> String
-showAddress addr = addr.street <> ", " <> addr.city <> ", " <> addr.state
+showAddress addr = addr.street <> ", " <> 
+                   addr.city <> ", " <> 
+                   addr.state
+-- ANCHOR_END: showAddress
 
+-- ANCHOR: showEntry_signature
 showEntry :: Entry -> String
-showEntry entry = entry.lastName <> ", " <> entry.firstName <> ": " <> showAddress entry.address
+-- ANCHOR_END: showEntry_signature
+-- ANCHOR: showEntry_implementation
+showEntry entry = entry.lastName <> ", " <> 
+                  entry.firstName <> ": " <> 
+                  showAddress entry.address
+-- ANCHOR_END: showEntry_implementation
 
+-- ANCHOR: emptyBook
 emptyBook :: AddressBook
 emptyBook = empty
+-- ANCHOR_END: emptyBook
 
+-- Note to reader: Delete this line. MAINTAINER CAUTION: There is an alternate copy of this implementation in the book source
```

Comment:
This caution exists because the initial implementation of this function as you work through the book is something like
```PureScript
insertEntry entry book = Cons entry book
```
and it later gets simplified to just what is seen below.  Similar story for the caution on line 60.  This seemed worth mentioning to the maintainer since changes to one of these implementations is likely going to also necessitate a change to the copied implementation in the book source.

Reply:
Good idea, although this wasn't clear to me when just reading that comment without seeing the rest of your explanation. I think this phrasing will make it easier for future maintainers to understand what to do:
```
NOTE TO MAINTAINER: If editing `insertEntry`, remember to also update the non-anchored (and unsimplified) version of this function that is hardcoded in the book text.
```

Or, after #370 is resolved:
```
This line should have been automatically deleted by resetSolutions.sh. See Chapter 2 for instructions. NOTE TO MAINTAINER: If editing `insertEntry`, remember to also update the non-anchored (and unsimplified) version of this function that is hardcoded in the book text.
```